src/main/java/co/ostorlab/ci/jenkins/connector/UploadInfo.java

```diff
@@ -34,11 +34,17 @@ public static UploadInfo fromJson(String json) throws JsonException {
         }
         //
         UploadInfo uploadInfo = new UploadInfo();
-        Integer scanId = Integer.parseInt((String) ((JsonObject) ((JsonObject) ((JsonObject) jsonObject.get("data")).get("createMobileScan")).get("scan")).get("id"));
-        uploadInfo.setScanId(scanId);
-
-        return uploadInfo;
-
+        if (jsonObject.get("data") != null &&
+                ((JsonObject) jsonObject.get("data")).get("createMobileScan") != null &&
+                ((JsonObject) ((JsonObject) jsonObject.get("data")).get("createMobileScan")).get("scan") != null &&
```

Comment:
I think u can use .getJsonObject("data") .. instead of .get and cast to JsonObject, can u give it a try ?